term/fhir-term/src/main/java/com/ibm/fhir/term/service/FHIRTermService.java

```diff
@@ -737,7 +737,17 @@ public ValidationOutcome validateCode(CodeSystem codeSystem, CodeableConcept cod
                 return outcome;
             }
         }
-        return validateCode(null, false, null);
+
+        StringBuilder message = new StringBuilder()
+                .append("None of the Coding values in the CodeableConcept were found to be valid in a CodeSystem with URL=")
+                .append(codeSystem.getUrl() == null ? null : codeSystem.getUrl().getValue())
+                .append(" and version=")
+                .append(codeSystem.getVersion() == null ? null : codeSystem.getVersion().getValue());
+
+        return ValidationOutcome.builder()
+                .result(Boolean.FALSE)
+                .message(string(message.toString()))
+                .build();
```

Comment:
This builder code is repeated multiple times - might make sense to have a generic buildValidationOutcome(Boolean result, String message, String display) method.

Reply:
ok, i followed this suggestion, but i created three overloaded flavors because i hate passing nulls around:
```
    private ValidationOutcome buildValidationOutcome(boolean result)
    private ValidationOutcome buildValidationOutcome(boolean result, java.lang.String message)
    private ValidationOutcome buildValidationOutcome(boolean result, java.lang.String message, LookupOutcome lookupOutcome)
```